### ISSUE TITLE
[question] Should XCRun use the conanfile's build env?

### ISSUE THREAD

**Issue by sfackler** [conan-io/conan#14297]
### What is your question?

I realized that #14195 doesn't fully solve the issues I was having when doing a Linux -> macOS cross build, since XCRun still assumes that `xcrun` is in the ambient `PATH`. I can work around it without too much trouble by just doing:

```python
with VirtualBuildEnv(self).vars().apply():
    fix_apple_shared_install_name(self)
```

but ideally the tool would apply the build env internally.

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @sfackler 

I just realized that ``XCRun()`` is not fully applying Conan profiles PATH env-var.
If something like this:

```
[buildenv]
PATH+=(path)/path/to/my/cross/xcrun
```
That would be a good solution, wouldn't it?

Note I don't think ``with VirtualBuildEnv(self).vars().apply():`` is a good solution, doesn't look great.

**Comment by sfackler** [comment]
I think you'd probably want the whole build env applied for things like LD_LIBRARY_PATH.

**Comment by memsharded** [comment]
Yes, exactly, I am trying to do https://github.com/conan-io/conan/pull/14326, so whatever you inject in the profile is used to run xcrun.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/14326 was merged, it will be in next 2.0.10 release

**Comment by sfackler** [comment]
Thanks!
